subscription/src/main/resources/org/killbill/billing/subscription/engine/dao/SubscriptionEventSqlDao.sql.stg

```diff
@@ -96,5 +96,22 @@ and is_active = '1'
 ;
 >>
 
-
-
+getActiveByAccountRecordId() ::= <<
+select
+<allTableFields("se.")>
+from <tableName()> se
+join (select
+      distinct(se1.subscription_id)
+      from <tableName()> se1
```

Comment:
Did you check the explain on that cross join? 😬

Reply:
I ran it for RH, seemed to be fine

Reply:
``` Limit  (cost=85322.04..85322.05 rows=3 width=245)
   ->  Sort  (cost=85322.04..85366.48 rows=17778 width=245)
         Sort Key: se.created_date
         ->  Hash Join  (cost=53772.12..85092.26 rows=17778 width=245)
               Hash Cond: ((se.subscription_id)::text = (se1.subscription_id)::text)
               ->  Seq Scan on subscription_events se  (cost=0.00..29932.70 rows=528540 width=245)
                     Filter: (account_record_id = 165)
               ->  Hash  (cost=53670.27..53670.27 rows=8148 width=37)
                     ->  Unique  (cost=53548.05..53588.79 rows=8148 width=37)
                           ->  Sort  (cost=53548.05..53568.42 rows=8148 width=37)
                                 Sort Key: se1.subscription_id
                                 ->  Gather  (cost=26465.82..53018.75 rows=8148 width=37)
                                       Workers Planned: 2
                                       ->  Parallel Hash Left Join  (cost=25465.82..51203.95 rows=3395 width=37)
                                             Hash Cond: ((se1.subscription_id)::text = (se2.subscription_id)::text)
                                             Filter: ((se2.id IS NULL) OR ((se2.effective_date)::timestamp without time zone > '2022-01-01 00:00:00'::timestamp without time zone))
                                             ->  Parallel Seq Scan on subscription_events se1  (cost=0.00..25364.35 rows=51117 width=37)
                                                   Filter: (is_active AND ((user_type)::text = 'CREATE'::text) AND (account_record_id = 165))
                                             ->  Parallel Hash  (cost=24602.96..24602.96 rows=69029 width=82)
                                                   ->  Parallel Seq Scan on subscription_events se2  (cost=0.00..24602.96 rows=69029 width=82)
                                                         Filter: (is_active AND ((user_type)::text = 'CANCEL'::text))```


Also rerun the query and it comes back immediately